client/dhcp_client_lwt.ml

```diff
@@ -0,0 +1,90 @@
+let src = Logs.Src.create "dhcp_client_lwt"
+module Log = (val Logs.src_log src : Logs.LOG)
+
+module Make(Time : Mirage_time_lwt.S) (Net : Mirage_net_lwt.S) = struct
+  open Lwt.Infix
+
+  type lease = Dhcp_wire.pkt
+
+  type t = lease Lwt_stream.t
+
+  let connect ?(with_xid)
+              ?(requests : Dhcp_wire.option_code list option) net =
```

Comment:
Just a minor comment here. Would it make sense to pass explicitly the mac address here (or as an optional argument) instead of relying on `Net.mac net` below? I have a use-case where this could be useful, but I am not sure if it is the right way to do it. Do we rely on `Net.mac` in other places of the stack?